doc: release_notes: skim down the release notes content (#82542)
zephyrproject-rtos/zephyr: doc/releases/release-notes-4.1.rst

:orphan:

..
  What goes here: removed/deprecated apis, new boards, new drivers, notable
  features. If you feel like something new can be useful to a user, put it
  under "Other Enhancements" in the first paragraph, if you feel like something
  is worth mentioning in the project media (release blog post, release
  livestream) put it under "Major enhancement".

  No list of bugfixes, minor changes, those are already in the git log, this is
  not a changelog.

  Does the entry have a link that contains the details? Just add the link, if
  you think it needs more details, put them in the content that shows up on the
  link.

  Are you thinking about generating this? Don't put anything at all.

  Does the thing require the user to change their application? Put it on the
  migration guide instead. (TODO: move the removed APIs section in the
  migration guide)

  Could the detailed paragraph that you are about to write go in the actual

>   project documentation instead and just be linked here? Do that instead.

Reviewer: this deserves being moved higher up IMO -- people sometimes don't realize that they end up writing really good "documentation" that just happens to get a lot less long-term visibility than it should

Author: done

  :c:member:`bt_conn_cb.disconnected` callback is invoked.

* :kconfig:option:`CONFIG_NATIVE_APPLICATION` has been deprecated.

* For the native_sim target :kconfig:option:`CONFIG_NATIVE_SIM_NATIVE_POSIX_COMPAT` has been
  switched to ``n`` by default, and this option has been deprecated.

* :kconfig:option:`CONFIG_BT_BUF_ACL_RX_COUNT`

* All HWMv1 board name aliases which were added as deprecated in v3.7 are now removed
  (:github:`82247`).

New APIs and options
====================

..
  Link to new APIs here, in a group if you think it's necessary, no need to get
  fancy just list the link, that should contain the documentation. If you feel
  like you need to add more details, add them in the API documentation code
  instead.

* Architectures

  * :kconfig:option:`CONFIG_ARCH_HAS_CUSTOM_CURRENT_IMPL`

  * :kconfig:option:`CONFIG_RISCV_CURRENT_VIA_GP`

* Crypto

  * :kconfig:option:`CONFIG_MBEDTLS_PSA_STATIC_KEY_SLOTS`

  * :kconfig:option:`CONFIG_MBEDTLS_PSA_KEY_SLOT_COUNT`

* Other

  * :kconfig:option:`CONFIG_BT_BUF_ACL_RX_COUNT_EXTRA`

  * :c:macro:`DT_ANY_INST_HAS_BOOL_STATUS_OKAY`

  * :c:struct:`led_dt_spec`.

New Boards & SoC
****************
..

>   Just link the board, detals go in the board description

Reviewer 2: `s/detals/details/` multiple times

Author: hey at least I was consistent

API Changes
***********

Removed APIs
============

>  * The Bluetooth HCI driver API has been removed. It has been replaced by a

Reviewer 3: I feel like this should still say something like "The old HCI driver API", since we still have an HCI driver API (described in the next sentence).

Author: yeah I was trying to remove some wording duplication, reading this back I made this sound super weird, changed it to "The legacy Bluetooth HCI..."

  :c:member:`bt_conn_cb.disconnected` callback is invoked.

* :kconfig:option:`CONFIG_NATIVE_APPLICATION` has been deprecated.

* For the native_sim target :kconfig:option:`CONFIG_NATIVE_SIM_NATIVE_POSIX_COMPAT` has been
  switched to ``n`` by default, and this option has been deprecated.

* :kconfig:option:`CONFIG_BT_BUF_ACL_RX_COUNT`

* All HWMv1 board name aliases which were added as deprecated in v3.7 are now removed
  (:github:`82247`).

New APIs and options
====================

..
  Link to new APIs here, in a group if you think it's necessary, no need to get
  fancy just list the link, that should contain the documentation. If you feel
  like you need to add more details, add them in the API documentation code
  instead.

* Architectures

  * :kconfig:option:`CONFIG_ARCH_HAS_CUSTOM_CURRENT_IMPL`
  * :kconfig:option:`CONFIG_RISCV_CURRENT_VIA_GP`

* Crypto

  * :kconfig:option:`CONFIG_MBEDTLS_PSA_STATIC_KEY_SLOTS`
  * :kconfig:option:`CONFIG_MBEDTLS_PSA_KEY_SLOT_COUNT`

* Other

  * :kconfig:option:`CONFIG_BT_BUF_ACL_RX_COUNT_EXTRA`
  * :c:macro:`DT_ANY_INST_HAS_BOOL_STATUS_OKAY`
  * :c:struct:`led_dt_spec`

New Boards & SoC
****************
..

>   Just link the board, details go in the board description

Reviewer: ```suggestion
  You may update this list as you contribute a new board during the release cycle, in order for it
  to be visible to people who might be looking at the working draft of the release notes. However,
  note that this list will be recomputed at the time of the release, so you don't *have* to update
  it.
  In any case, just link the board, details go in the board description.
```

Author: done

* :zephyr:board:`max78002evkit`
* :zephyr:board:`acp_6_0_adsp`
* :zephyr:board:`ucan`
* :zephyr:board:`ttgo_t7v1_5`
* :zephyr:board:`ttgo_t8s3`
* :zephyr:board:`m5stack_cores3`
* :zephyr:board:`mks_canable_v20`
* :zephyr:board:`octopus_io_board`
* :zephyr:board:`octopus_som`
* :zephyr:board:`candlelight`
* :zephyr:board:`candlelightfd`
* :zephyr:board:`xiao_esp32c6`
* :zephyr:board:`rp2040_zero`
* :zephyr:board:`ch32v003evt`
* :zephyr:board:`we_orthosie1ev`
* :zephyr:board:`mini_stm32h7b0`

New Drivers
***********
..

>   Just link the driver, details go in the binding description

Reviewer: Assuming we'll do the same here: 

```suggestion
    Same as above for boards, this will also be recomputed at the time of the release.
    Just link the driver, details go in the binding description
```

Author: done

* :dtcompatible:`ti,tca9544a`
* :dtcompatible:`ti,tca9544a-channel`
* :dtcompatible:`ite,it8801-kbd`
* :dtcompatible:`wch,pfic`
* :dtcompatible:`linaro,ivshmem-mbox`
* :dtcompatible:`renesas,ra-mdio`
* :dtcompatible:`awinic,aw9523b`
* :dtcompatible:`ite,it8801-altctrl`
* :dtcompatible:`ite,it8801-mfd-map`
* :dtcompatible:`ite,it8801-mfd`
* :dtcompatible:`nordic,npm2100`
* :dtcompatible:`nordic,nrf-ppib`
* :dtcompatible:`fujitsu,mb85rsxx`
* :dtcompatible:`silabs,dbus-pinctrl`
* :dtcompatible:`wch,afio`
* :dtcompatible:`ite,it8801-pwm`
* :dtcompatible:`nordic,npm2100-regulator`
* :dtcompatible:`adi,adxl366`
* :dtcompatible:`adi,adxl366`
* :dtcompatible:`hc-sr04`
* :dtcompatible:`nordic,npm2100-vbat`
* :dtcompatible:`sensirion,scd40`
* :dtcompatible:`sensirion,scd41`
* :dtcompatible:`sensirion,sts4x`
* :dtcompatible:`wch,usart`
* :dtcompatible:`ite,it8xxx2-spi`
* :dtcompatible:`renesas,ra-spi`
* :dtcompatible:`vnd,i2s`
* :dtcompatible:`mediatek,ostimer64`
* :dtcompatible:`wch,systick`
* :dtcompatible:`ambiq,usb`
* :dtcompatible:`zephyr,video-emul-imager`
* :dtcompatible:`zephyr,video-emul-rx`
* :dtcompatible:`nordic,npm2100-wdt`

New Samples
***********

..

>   Just link the samples, details go in the sample documentation itself

Reviewer: ```suggestion
    Same as above for boards and drivers, this will also be recomputed at the time of the release.
    Just link the sample, details go in the sample documentation itself.
```

Author: done
